packages/strapi-admin/services/permission/permissions-manager/query-builers.js

```diff
@@ -1,18 +1,19 @@
 'use strict';
 
 const _ = require('lodash');
+const { get, has } = require('lodash/fp');
 const { rulesToQuery } = require('@casl/ability/extra');
 const { VALID_REST_OPERATORS } = require('strapi-utils');
 
 const ops = {
   common: VALID_REST_OPERATORS.map(op => `$${op}`),
-  boolean: ['$or'],
+  boolean: ['$or', '$and'],
   cleanable: ['$elemMatch'],
 };
 
 const buildCaslQuery = (ability, action, model) => {
   const query = rulesToQuery(ability, action, model, o => o.conditions);
-  return query && _.has(query, '$or') ? _.pick(query, '$or') : {};
+  return query && has('$or', query) ? get('$or[0].$and', query) : {};
```

Comment:
why not keep pick here ?  by the way the _.has is unnecessary as pick returns and empty object if the query is undefined or if the key doesn't exist

Reply:
Because previously we needed the `or` (every condition was built using an `orWhere`).
Here we could keep the `or` + nested `and` & conditions but it would complexity the final query (in db) as we would have a useless `or` & `and` clause built (`orWhere(where(conditionsWheres))` instead of `conditionsWheres` only).

Reply:
Sounds good. You can at least just return the get only as it handles the two previous conditions in your ternary

Reply:
Yeah for sure